sdk/ml/azure-ai-ml/azure/ai/ml/operations/_environment_operations.py

```diff
@@ -92,6 +93,37 @@ def create_or_update(self, environment: Environment) -> Environment:
                 )
             sas_uri = None
             if self._registry_name:
+                if isinstance(environment, WorkspaceAssetReference):
+                    # verify that environment is not already in registry
+                    try:
+                        self._version_operations.get(
+                            name=environment.name,
+                            version=environment.version,
+                            resource_group_name=self._resource_group_name,
+                            registry_name=self._registry_name,
+                        )
+                    except Exception as err:  # pylint: disable=broad-except
+                        if isinstance(err, ResourceNotFoundError):
+                            pass
+                        else:
+                            raise err
+                    else:
+                        msg = "A environment with this name and version already exists in registry"
+                        raise ValidationException(
+                            message=msg,
+                            no_personal_data_message=msg,
+                            error_target=ErrorTarget.ENVIRONMENT,
```

Comment:
`error_target` does not exist. it is `target`

Reply:
I fixed this here https://github.com/Azure/azure-sdk-for-python/pull/28585/files in `_model_operations.py`, you may need to sync with main branch.